documentation/API-authentication-token.md

```diff
@@ -1,12 +1,19 @@
 # API authentication token
 
-< to be described >
+When requesting a token on the oauth2 token resource using the client_credentials a queryparameter containing the identifier for the school. A scope to access data can optionally be requested and if consent has been given, will be added as a scope in the token defined below.
```

Comment:
The OAuth 2.0 specification has the concept of "extension grants" (https://datatracker.ietf.org/doc/html/rfc6749#section-4.5). This can be used to formalize the use of the parameter digideliveryid in the client credentials flow. 

An example of a token request using a digideliveryid extension grant could be:

```
POST /token HTTP/1.1
Host: server.example.com
Content-Type: application/x-www-form-urlencoded

grant_type=client_credentials_digideliveryid&client_id=[CLIENT_ID]&client_secret=[CLIENT_SECRET]&digideliveryid=[DIGI_DELIVERY_ID]
```


And if the token request was successful, the token response could be:

```
{
  "access_token":"[ACCESS_TOKEN]",
  "token_type":"Bearer",
  "expires_in":86400,
  "digideliveryid":"[DIGI_DELIVERY_ID]"
}
```

Reply:
Hi Arjen, leuk je weer tegen te komen. Lets drink a virtual coffee Edwin V kan je wel mijn contact info geven ;)

Reply:
If we go for a custom grant type, does that then limit the use of libraries to do the authentication handshakes? I prefer to use libraries which are tested by more people/companies, then having to a custom implementation.
Is it also an option to pass the school info using the acr_values parameter?

Reply:
Please check my remark on #26 about parentID.

Reply:
I agree with Henk on this one. If it limits the libraries which we can use, we shouldnt make it custom

Reply:
The suggested client credentials flow is defined in the OAuth 2.0 RFC. The parameter acr_values is defined in the OpenID Connect Core specification as a parameter for an authorise request in the authorization code flow (section 3.1.2.1). The parameter acr_values could be used in the client credentials flow, but is far from standard.

Besides that, I can imagine that adding acr_values to the client credentials flow could lead to comparable limitations in OAuth 2.0 libraries.

Reply:
And the client credentials flow is so simple you could wonder if you really need an OAuth library for that. Auth0 has some sample implementations of a client credentials token request in multiple languages: https://auth0.com/docs/get-started/authentication-and-authorization-flow/call-your-api-using-the-client-credentials-flow.